File: web/src/services/api/generation-history.ts
import { apiDelete, apiGet, apiPost, compactApiParams } from "@/services/api/request";

export type GenerationHistoryType = "image" | "video";

export type GenerationHistoryMedia = {
    cloudFileId: string;
    storageKey: string;
    url: string;
    fileType: GenerationHistoryType;
    contentType: string;
    size: number;
    width: number;
    height: number;
    expiresAt: string;
};

export type GenerationHistoryReference = {
    name: string;
    type: string;
    url: string;
    storageKey: string;
};

export type GenerationHistory = {
    id: string;
    userId: string;
    type: GenerationHistoryType;
    title: string;
    prompt: string;
    model: string;
    config: Record<string, string>;
    references: GenerationHistoryReference[];
    media: GenerationHistoryMedia[];
    status: "成功" | "失败";
    error: string;
    durationMs: number;
    expiresAt: string;
    createdAt: string;
    updatedAt: string;
};

export type GenerationHistoryList = {
    items: GenerationHistory[];
    total: number;
};

export type SaveGenerationHistoryInput = Pick<GenerationHistory, "type" | "title" | "prompt" | "model" | "config" | "references" | "media" | "status" | "error" | "durationMs">;

export async function fetchGenerationHistories(token: string, query: { type: GenerationHistoryType; page?: number; pageSize?: number }) {
    return apiGet<GenerationHistoryList>("/api/v1/generation-histories", compactApiParams(query), token);
}

export async function saveGenerationHistory(token: string, input: SaveGenerationHistoryInput) {
    return apiPost<GenerationHistory>("/api/v1/generation-histories", input, token);
}

export async function deleteGenerationHistory(token: string, id: string) {
    return apiDelete<boolean>(`/api/v1/generation-histories/${encodeURIComponent(id)}`, token);
}

File: model/generation_history.go
package model

type GenerationHistoryType string

const (
	GenerationHistoryTypeImage GenerationHistoryType = "image"
	GenerationHistoryTypeVideo GenerationHistoryType = "video"
)

type GenerationHistoryMedia struct {
	CloudFileID string        `json:"cloudFileId"`
	StorageKey  string        `json:"storageKey"`
	URL         string        `json:"url"`
	FileType    CloudFileType `json:"fileType"`
	ContentType string        `json:"contentType"`
	Size        int64         `json:"size"`
	Width       int           `json:"width"`
	Height      int           `json:"height"`
	ExpiresAt   string        `json:"expiresAt"`
}

type GenerationHistoryReference struct {
	Name       string `json:"name"`
	Type       string `json:"type"`
	URL        string `json:"url"`
	StorageKey string `json:"storageKey"`
}

// GenerationHistory 记录用户生成历史。媒体文件生命周期跟随关联的 cloud_files。
type GenerationHistory struct {
	ID         string                       `json:"id" gorm:"primaryKey"`
	UserID     string                       `json:"userId" gorm:"index"`
	Type       GenerationHistoryType        `json:"type" gorm:"index"`
	Title      string                       `json:"title"`
	Prompt     string                       `json:"prompt"`
	Model      string                       `json:"model"`
	Config     map[string]string            `json:"config" gorm:"serializer:json"`
	References []GenerationHistoryReference `json:"references" gorm:"serializer:json"`
	Media      []GenerationHistoryMedia     `json:"media" gorm:"serializer:json"`
	Status     string                       `json:"status"`
	Error      string                       `json:"error"`
	DurationMs int64                        `json:"durationMs"`
	ExpiresAt  string                       `json:"expiresAt" gorm:"index"`
	CreatedAt  string                       `json:"createdAt"`
	UpdatedAt  string                       `json:"updatedAt"`
}

type GenerationHistoryList struct {
	Items []GenerationHistory `json:"items"`
	Total int                 `json:"total"`
}
